pkg/cache/clusterqueue_snapshot.go

```diff
@@ -207,14 +207,14 @@ type WorkloadTASRequests map[kueue.ResourceFlavorReference]FlavorTASRequests
 
 func (c *ClusterQueueSnapshot) FindTopologyAssignmentsForWorkload(
 	tasRequestsByFlavor WorkloadTASRequests,
-	simulateEmpty bool) TASAssignmentsResult {
+	simulateEmpty bool, wl *kueue.Workload) TASAssignmentsResult {
```

Comment:
Instead of having another required parameter, and passing the entire workload I would prefer to just pass the annotation value. 

I think it would be cleaner to use the options pattern. When we could call it. `FindTopologyAssignmentsForWorkload(tasRequests, WithSimulateEmpty(), WithNodeToReplace(<nodeName>)).`. And in the simplest cases, one would just call: `FindTopologyAssignmentsForWorkload(tasRequests)`.

However, I see this is late in the release cycle I'm totally ok with a follow up. wdyt?

Reply:
I like the idea, will do in a follow up